app/drivers/sensor/ec11/ec11.c

```diff
@@ -59,23 +61,38 @@ static int ec11_sample_fetch(const struct device *dev, enum sensor_channel chan)
     drv_data->pulses += delta;
     drv_data->ab_state = val;
 
-    drv_data->ticks = drv_data->pulses / drv_cfg->resolution;
-    drv_data->delta = delta;
-    drv_data->pulses %= drv_cfg->resolution;
+    if (drv_cfg->steps == 0) {
+        drv_data->ticks = drv_data->pulses / drv_cfg->resolution;
+        drv_data->delta = delta;
+        drv_data->pulses %= drv_cfg->resolution;
+    }
 
     return 0;
 }
 
 static int ec11_channel_get(const struct device *dev, enum sensor_channel chan,
                             struct sensor_value *val) {
     struct ec11_data *drv_data = dev->data;
+    const struct ec11_config *drv_cfg = dev->config;
+    int32_t pulses = drv_data->pulses;
 
     if (chan != SENSOR_CHAN_ROTATION) {
         return -ENOTSUP;
     }
 
-    val->val1 = drv_data->ticks;
-    val->val2 = drv_data->delta;
+    drv_data->pulses = 0;
+
+    if (drv_cfg->steps > 0) {
+        val->val1 = (pulses * FULL_ROTATION) / drv_cfg->steps;
+        val->val2 = (pulses * FULL_ROTATION) - (val->val1 * drv_cfg->steps);
```

Comment:
It took me a while to figure out that this is just getting the remainder. Consider switching to
```suggestion
        val->val2 = (pulses * FULL_ROTATION) % drv_cfg->steps;
```
for better readability. 

Doing a division gets you the remainder for free, so this gets optimized to a single operation either way you write it: https://godbolt.org/z/Y9rcdP844